File: src/test/java/com/rafaelfiume/tictactoe/matchers/EmptyBoardMatcher.java
package com.rafaelfiume.tictactoe.matchers;

import static java.lang.System.lineSeparator;

public class EmptyBoardMatcher extends BoardMatcher {

    // Duplicated? Yes! That ensures we are not making a mistake asserting a wrong board
    private static final String EXPECTED_BOARD = "" +
            "   |   |   " + lineSeparator() +
            "---+---+---" + lineSeparator() +
            "   |   |   " + lineSeparator() +
            "---+---+---" + lineSeparator() +
            "   |   |   ";


    public EmptyBoardMatcher() {
        super(EXPECTED_BOARD);
    }

}

File: src/test/java/com/rafaelfiume/tictactoe/matchers/EmptyBoardMatcher.kt
package com.rafaelfiume.tictactoe.matchers

import java.lang.System.lineSeparator

class EmptyBoardMatcher : BoardMatcher(EmptyBoardMatcher.EXPECTED_BOARD) {

    companion object {

        private val EXPECTED_BOARD = "" +
                "   |   |   " + lineSeparator() +
                "---+---+---" + lineSeparator() +
                "   |   |   " + lineSeparator() +
                "---+---+---" + lineSeparator() +
                "   |   |   "

        @JvmStatic fun anEmptyBoard() = EmptyBoardMatcher()
    }

}
